compiler/injectdestructors.nim

```diff
@@ -430,7 +430,9 @@ proc passCopyToSink(n: PNode; c: var Con; s: var Scope): PNode =
   if hasDestructor(c, n.typ):
     let typ = n.typ.skipTypes({tyGenericInst, tyAlias, tySink})
     let op = getAttachedOp(c.graph, typ, attachedDup)
-    if op != nil and tfHasOwned notin typ.flags:
+    if op != nil and tfHasOwned notin typ.flags and (
+          sfOverriden in op.flags or
+          typ.skipTypes({tyDistinct}).kind notin {tyObject, tyTuple, tyArray, tySequence}):
```

Comment:
Just skipping `tyDistinct` might not be enough because the base of the distinct type could also be `tyGenericInst`, `tyAlias` etc